File: src/modifiers/cornerRadius.ts
import { createModifier } from './createModifier';

export interface CornerRadiusParams {
  /// Corner radius in points.
  radius: number;
}

/// SwiftUI `.cornerRadius(_:)`. Clips the view to a rounded rectangle of the
/// given corner radius. Deprecated in newer SDKs in favor of
/// `.clipShape(RoundedRectangle(...))`, but still available on watchOS 9+.
export function cornerRadius(
  params: CornerRadiusParams | number
): ReturnType<typeof createModifier> {
  if (typeof params === 'number') return createModifier('cornerRadius', { radius: params });
  return createModifier('cornerRadius', params);
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/CornerRadius.swift
import SwiftUI

/// SwiftUI `.cornerRadius(_:)`. Clips the view to a rounded rectangle of the
/// given `radius`. The API is deprecated in newer SDKs (in favor of
/// `.clipShape(RoundedRectangle(...))`) but remains available on watchOS 9+.
enum RNWCornerRadiusModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("cornerRadius") { view, params, _ in
            let radius = params.cgFloat("radius") ?? 0
            return AnyView(view.cornerRadius(radius))
        }
    }
}
